src/entities/calendaritem.cpp

```diff
@@ -77,15 +79,21 @@ void CalendarItem::setAlarmDate(const QDateTime &dateTime) { this->alarmDate = d
 
 void CalendarItem::setPriority(int value) { this->priority = value; }
 
-void CalendarItem::setCompleted(bool value) { this->completed = value; }
+void CalendarItem::setProgress(int value) { this->progress = value; }
```

Comment:
You need to use `setCompleted(value = 100)` too here! But don't use `setProgress` inside `setCompleted` then, but set the value directly. 😉

Reply:
Hm, but you don't want to override progress too in that case inside setCompleted... you need to check that somewhere else...

Reply:
Yeah, I wasn't sure which way to go here. :sweat_smile:

Reply:
Either way, progress and completed need to be "linked" in some way.